lock-api/src/main/java/com/palantir/lock/client/RequestBatchers.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import com.palantir.atlasdb.timelock.api.Namespace;
+import com.palantir.lock.watch.LockWatchEventCache;
+import com.palantir.lock.watch.NoOpLockWatchEventCache;
+import com.palantir.lock.watch.StartTransactionsLockWatchEventCache;
+import java.util.Optional;
+
+public final class RequestBatchers {
+    private final LockWatchEventCache lockWatchEventCache;
+    private final StartTransactionsLockWatchEventCache startTransactionsLockWatchEventCache;
+    private final Optional<Namespace> namespace;
+    private final Optional<MultiClientTransactionStarter> maybeTransactionStarter;
+    private final Optional<MultiClientCommitTimestampGetter> maybeCommitTimestampGetter;
```

Comment:
Is there a way of avoiding this? / would we expect correlations? It feels like either all 3 will be present or absent - if that's the case it's probably better to have an `Optional<T>` where `T` is some type that covers all of these three objects.

Reply:
Looking at the usage it seems you want a non-optional namespace (just create a fake one for the tests) and an optional of the 2 multi-client objects.